Rules/UseConsistentWhitespace.cs

```diff
@@ -530,6 +533,29 @@ private IEnumerable<DiagnosticRecord> FindOperatorViolations(TokenOperations tok
                     continue;
                 }
 
+                // exclude assignment operator inside of multi-line hash tables if requested
+                if (IgnoreAssignmentOperatorInsideHashTable && tokenNode.Value.Kind == TokenKind.Equals)
+                {
+                    var enclosingHashTables = tokenOperations.Ast.FindAll(a => a.Extent.StartOffset <= tokenNode.Value.Extent.StartOffset && a.Extent.EndOffset >= tokenNode.Value.Extent.EndOffset && a is HashtableAst, true);
+                    if (enclosingHashTables.Count() > 0)
+                    {
+                        Ast innermostEnclosingHashTable = enclosingHashTables.First();
+                        int smallestSizeSoFar = int.MaxValue;
+                        foreach (var hashTable in enclosingHashTables){
+                            int currentHashTableSize = hashTable.Extent.EndOffset - hashTable.Extent.StartOffset;
+                            if (currentHashTableSize < smallestSizeSoFar)
+                            {
+                                innermostEnclosingHashTable = hashTable;
+                                smallestSizeSoFar = currentHashTableSize;
+                            }
+                        }
+                        if (innermostEnclosingHashTable.Extent.StartLineNumber != innermostEnclosingHashTable.Extent.EndLineNumber)
+                        {
+                            continue;
+                        }
+                    }
+                }
```

Comment:
So with this strategy, we collect all the hashtable ASTs that contain the token offset, and then try to find the smallest one. That works, but I think a better strategy is to:

- Find the smallest AST that contains the token offset
- Test whether that AST is a `HashtableAst` (since the `=` in HashtableAst key-value pairs is directly subordinate to that AST)

That way there's no need to sift through multiple ASTs reflecting on them to work out whether they're `HashtableAst`s, only the one containing the `=` operator.

One way to find the smallest containing AST is to do as you've done with the `FindAll()` method and sift through the returned ASTs looking for the one with the smallest extent. A slightly more efficient way is to implement a visitor like the one in https://github.com/PowerShell/vscode-powershell/issues/2860#issuecomment-669506615.